### ISSUE TITLE
trusted proxies does not work as intended

### ISSUE THREAD

**Issue by 2020Sanoj** [nextcloud/all-in-one#828]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Install nextcloud aio with custom apache port
2. Nextcloud brute force thinks every access comes from ::1 and the log thinks every wrong login comes from ::1 / `docker logs nextcloud-aio-nextcloud` shows 172.25.0.9
3. Set ::1 as trusted proxy in the config.php
4. With apache and caddy fd00:dead:beef:c0::1 is now shown / with the Nginx Proxy Manager 172.17.0.1, both are the gateways of the docker bridge network (X-Forwarded-For header is set in all the configurations)
5. Setting fd00:dead:beef:c0::1/172.17.0.1/172.25.0.9 as trusted proxy changes nothing.

### Expected behavior 
The real IP is shown (which is sent in the X-Forwarded-For header)
### Actual behavior 
::1 is shown as IP to access / `docker logs nextcloud-aio-nextcloud` shows 172.25.0.9
With ::1 set as trusted proxy, it shows fd00:dead:beef:c0::1  /  `docker logs nextcloud-aio-nextcloud` shows 172.25.0.9
Setting fd00:dead:beef:c0::1/172.17.0.1/172.25.0.9 as trusted proxy changes nothing
This makes the brute force protection useless and makes it block everything, instead of a certain IP
If my cloud gets blocked, I need to run `[sudo] docker exec -it nextcloud-aio-nextcloud php occ security:bruteforce:reset ::1` to make it possible for me to login, (Whitelisting ::1 in brute force settings app also works)

### Host OS 
Debian GNU/Linux 11 (bullseye)

#### Nextcloud AIO version 
v1.4.0  and v1.4.1
#### Current channel 
Beta
#### Other valuable info 
Default Nextcloud config.php (Apache/Caddy and NPM):
![grafik](https://user-images.githubusercontent.com/75573284/173386275-dd351800-f707-45ff-8067-8e5ba61b6dcf.png)
NPM with ::1 as set trusted proxy:
![grafik](https://user-images.githubusercontent.com/75573284/173386466-c43615ad-f1c1-46ac-8db9-6579e4b7a1fc.png)
Apache2/Caddy with ::1 set as trusted proxy:
![grafik](https://user-images.githubusercontent.com/75573284/173386505-337054d4-f62f-4ee9-a054-0cf65b117094.png)

fd00:dead:beef:c0::1 and 172.17.0.1 are the gateways of the docker bridge network and 172.25.0.9 the IP of the nextcloud-aio-apache container
The X-Forwarded-For header is set by all three proxies to my real IP (`185.***.**.**`)

Edit:
Also, this block of trusted proxies does not work (And yes, some entries in this block are nonsense)
```
  array (
    0 => '127.0.0.1',
    1 => '172.25.0.9:*',
    2 => '172.17.0.1:*',
    3 => '0.0.0.0/0:*',
    4 => '0.0.0.0/0',
    5 => '0.0.0.0',
    6 => '::1/0:*',
    7 => '::1/0',
    8 => '::1',
  ),
```

**Comment by szaimen** [comment]
@2020Sanoj can you try to add `trusted_proxies private_ranges` to the /Caddyfile in the apache container, restart it and see if it helps?

**Comment by 2020Sanoj** [comment]
I've did both and now my real IP is shown

**Comment by 2020Sanoj** [comment]
![grafik](https://user-images.githubusercontent.com/75573284/173442083-30f2ec80-ff29-4632-9f6a-9596d7ba2f2f.png)

**Comment by 2020Sanoj** [comment]
```
    # Nextcloud
    route {
        rewrite /.well-known/carddav /remote.php/dav
        rewrite /.well-known/caldav /remote.php/dav
        header Strict-Transport-Security max-age=31536000;
        reverse_proxy localhost:8000 {
         trusted_proxies private_ranges
        }
    }
```
I've changed the nextcloud part of the Caddyfile to this, but maybe this also need to be added to all other reverse proxies, or?

**Comment by szaimen** [comment]
> I've changed the nextcloud part of the Caddyfile to this, but maybe this also need to be added to all other reverse proxies, or?

How many reverse proxies are there? if only caddy inside the container and nginx proxy manager outside, I think this should be fine...

**Comment by 2020Sanoj** [comment]
With "all other reverse proxies", I meant for example https://github.com/nextcloud/all-in-one/blob/145d83c0006dc6a589ddb7d46c7357364d673fb1/Containers/apache/Caddyfile#L14 or https://github.com/nextcloud/all-in-one/blob/145d83c0006dc6a589ddb7d46c7357364d673fb1/Containers/apache/Caddyfile#L20

**Comment by 2020Sanoj** [comment]
So that this needs to be added to every proxy inside the caddyfile

**Comment by 2020Sanoj** [comment]
But this only works if `::1` is set as trusted proxy

**Comment by 2020Sanoj** [comment]
Tomorrow I will test, if it is possible to use instead of `127.0.0.1` AND `::1` as trusted proxy only `::1` or only `localhost`.
But setting both `127.0.0.1` and `::1` as trusted proxy works.

**Comment by 2020Sanoj** [comment]
It only works if `127.0.0.1` and `::1` are set as trusted proxy

**Comment by crocobc** [comment]
Sorry for this stupid question but i'm newbie in docker....

I search before asking...

What are the exact command to do these steps ?

@2020Sanoj can you try to add trusted_proxies private_ranges to the /Caddyfile in the apache container, restart it and see if it helps?

Another thing that you may try if the above does not work is doing this inside the apache container:
https://github.com/nextcloud/docker/blob/00d88733d0d93d0abc628c7b18bc589181c34cbb/24/apache/Dockerfile#L116-L123

If i go inside the container "sudo docker exec -it nextcloud-aio-apache  bin/bash" no permission to edit file (nor vi or nano)
Same problem with the command in the above link  (got permission denied when trying docker exec)

YES i'm a BIG newbie. Sorry

Thank you in advance.

Regards

**Comment by 2020Sanoj** [comment]
I think 2. Would be a security risk and then the custom apache port should only be exposed to localhost

**Comment by szaimen** [comment]
Hello, this should be fixed with 1.4.2 Beta.
